### PR TITLE
chore(ci): Enable node_modules & ccache caching

### PR DESCRIPTION
**Pull request by jhermsmeier**
This enables caching of `node_modules`, and `ccache` directories,
as well as installation of ccache on Mac OS.

Change-Type: patch

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -39,6 +44,8 @@ before_install:
     else
       export HOST_OS="$TRAVIS_OS_NAME";
     fi
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then brew install ccache; fi
```
Comment (jviotti) [inline .travis.yml]: Could we move this to `scripts/ci/install.sh`?

File: .travis.yml [hunk 2]
```diff
@@ -39,6 +44,8 @@ before_install:
...
+  - if [[ "$TRAVIS_OS_NAME" == "osx" ]]; then PATH=/usr/local/opt/ccache/libexec:$PATH; fi
```
Comment (jviotti) [inline .travis.yml]: Would it do any harm if we add this in the `env` section?

File: .travis.yml (hunk 1 above)
Comment (jhermsmeier) [inline .travis.yml]: Sure, done

File: .travis.yml (hunk 2 above)
Comment (jhermsmeier) [inline .travis.yml]: I moved it up now, not sure which env section exactly you're referring to?

Comment (jviotti) [inline .travis.yml]: I mean this: https://github.com/resin-io/etcher/blob/master/.travis.yml#L17

Comment (jhermsmeier) [inline .travis.yml]: Oh, well, this is Mac OS only though...

Comment (jviotti) [inline .travis.yml]: Fair enough. Why is this needed though? Shouldn't `brew` take care of symlinking any binary to `/usr/local/bin`?

Comment (jhermsmeier) [inline .travis.yml]: I was thinking the same, but
```
$ brew info ccache
[...]
==> Caveats
To install symlinks for compilers that will automatically use
ccache, prepend this directory to your PATH:
  /usr/local/opt/ccache/libexec
```

Comment (jviotti) [inline .travis.yml]: Bah, fair enough, I'll approve :)

**Comment by lurch** [comment]
Does this mean that the Travis Linux images come with ccache pre-installed, but the Travis OSX images don't? Do we need to do anything special to get ccache and node_modules caching working inside the Docker images too?

File: .travis.yml (hunk 2 above)
Comment (lurch) [inline .travis.yml]: I've also just found https://docs.travis-ci.com/user/caching/#ccache-cache

**Comment by jhermsmeier** [comment]
I guess we could map the `node_modules` into the docker containers, but that'd probably mess up the cache every time... and the caches are meant for the travis box itself, so I'm not sure we should do that.

**Comment by lurch** [comment]
@jhermsmeier Any idea why the Travis builds are failing in #1558 ? They're printing messages like:
```
single-line-log@1.1.2 /etcher/node_modules/electron/node_modules/single-line-log
semver@5.3.0 /etcher/nochange detected (content changed, file is created, or file is deleted):
/home/travis/build/resin-io/etcher/node_modules/lzma-native/build/liblzma/build/lib/liblzma.a
/home/travis/build/resin-io/etcher/node_modules/lzma-native/build/liblzma/config.log
/home/travis/build/resin-io/etcher/node_modules/lzma-native/build/liblzma/libtool
/home/travis/build/resin-io/etcher/node_modules/lzma-native/build/liblzma/src/liblzma/.libs/liblzma.a
changes detected, packing new archive
.
.
.
.
FAILED: tar -Pzcf /home/travis/.casher/push.tgz /home/travis/build/resin-io/etcher/node_modules
tar: /home/travis/build/resin-io/etcher/node_modules: file changed as we read it
uploading archive
Done. Your build exited with 1.
```

**Comment by jhermsmeier** [comment]
@lurch why did the one where it got merged go green then? https://travis-ci.org/resin-io/etcher/builds/247705868 Only the PR that followed that into master went red... Something doesn't add up :/

**Comment by jhermsmeier** [comment]
Also not sure why the `node_modules` would still be modified at that point in time

**Comment by jhermsmeier** [comment]
Looks like the cache builder starts while `$ ./scripts/ci/build-installers.sh -o $HOST_OS -r $TARGET_ARCH` is still running, from what I can tell
